hoot-core/src/main/cpp/hoot/core/io/OsmChangeWriter.cpp

```diff
@@ -40,56 +44,50 @@ void OsmChangeWriter::write(QIODevice &d, ChangeSetProviderPtr cs)
   writer.writeAttribute("version", "0.6");
   writer.writeAttribute("generator", "hootenanny");
 
-  Change::ChangeType last = Change::Unknown;
-
+  //Make the changeset have only one set of entries per change type.  Also for readability,
+  //sort the set of changes within each change type by element type.
+  QMap<Change::ChangeType, QMap<ElementType::Type, QVector<ConstElementPtr> > > elementsByChangesetTypeAndElementType;
```

Comment:
This change makes the OsmChangeWriter memory bound. In its previous implementation hoot could derive changes of two sorted inputs w/o being memory bound.